hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -787,11 +789,35 @@ def update_shrink_target(self, new_target):
             self.shrinks += 1
             if (
                 len(new_target.blocks) != len(self.shrink_target.blocks)
-                or self.all_block_bounds(new_target) != self.all_block_bounds()
+                or new_target.blocks.endpoints != self.shrink_target.blocks.endpoints
```

Comment:
Instead of exposing `endpoints` directly, this could potentially be some kind of `Blocks.has_same_endpoints` method, so that the endpoints remain an internal detail.

Reply:
It could, but I wasn't sure it was worth it - *all* of this code is internal and fairly tightly coupled.